sdk/iOS/test/MSSyncTableTests.m

```diff
@@ -1082,9 +1083,348 @@ -(void) testPushNetworkTimeoutWithErroredOps
     XCTAssertEqual(filteredClient.syncContext.pendingOperationsCount, 3);
 }
 
+// Tests cancellation of the push operation at various stages of a push workflow.
+// - Verifies that a push following a cancelled push operation works as expected
+// - Verifies that cancelling a push operation finishes the push NSOperation regardless of the cancellation point
+// - Verifies that the expected number of items are pushed to the server
+-(void) testPushCancellability
```

Comment:
I agree in general it is non-ideal to couple multiple tests. However, in this case, we are essentially just testing cancel. The 3 verification bullets listed are all attributes of a clean cancel and not really different test cases.